src/screens/Manager/Manager.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useCallback, useEffect } from "react";
+import { NavigationActions } from "react-navigation";
+import type { Action, State } from "@ledgerhq/live-common/lib/apps";
+
+import { ManagerContext } from "./shared";
+import AppsScreen from "./AppsScreen";
+import GenericErrorBottomModal from "../../components/GenericErrorBottomModal";
+import QuitManagerModal from "./Modals/QuitManagerModal";
+import StorageWarningModal from "./Modals/StorageWarningModal";
+import AppDependenciesModal from "./Modals/AppDependenciesModal";
+import UninstallDependenciesModal from "./Modals/UninstallDependenciesModal";
+
+type Props = {
+  screenProps: {
+    state: State,
+    dispatch: Action => void,
+  },
+  navigation: *,
+};
+
+export default ({ screenProps: { state, dispatch }, navigation }: Props) => {
+  const { apps, currentError, installQueue, uninstallQueue } = state;
+  const [quitManagerAction, setQuitManagerAction] = useState(false);
+
+  useEffect(() => {
+    const blockNavigation = installQueue.length + uninstallQueue.length > 0;
```

Comment:
this blockNavigation var can be moved in the render function, that way you inject it as a dep of useEffect and it will be re-triggered less time.